tests/controllers/pagamento.controller.test.ts

```diff
@@ -0,0 +1,442 @@
+              .fn()
+              .mockRejectedValue(new Error("Failed to create"));
+
+          await pagamentoController.post(
+              mockRequest,
+              mockResponse,
+              mockNextFunction,
+          );
+
+          expect(mockNextFunction).toHaveBeenCalledWith(
+              expect.any(Error),
+          );
+      });
+    });
+  });
+
+  describe("Given getAll method is called", () => {
+    describe("Just return all pagamentos", () => {
+      it("should return all pagamentos", async () => {
+        const mockRequest = {} as any;
+
+        const mockResponse = {
+          status: jest.fn().mockReturnThis(),
+          json: jest.fn(),
+        } as any;
+
+        const mockNextFunction = jest.fn();
+        const expectedResult = [
+          { id: "someId", pagamentoId: "someOrderId", valorTotal: 20, tipo: PagamentoTipoEnum.Aprovado},
+          { id: "someAnotherId", pagamentoId: "someAnotherOrderId", valorTotal: 39, tipo: PagamentoTipoEnum.Recusado},
+         ];
+
+        pagamentoUseCaseMock.getAll = jest
+          .fn()
+          .mockResolvedValue(expectedResult);
+
+          await pagamentoController.get(
+            mockRequest,
+            mockResponse,
+            mockNextFunction,
+          );
+
+          expect(mockResponse.status).toHaveBeenCalledWith(
+            StatusCode.ok,
+          );
+          expect(mockResponse.json).toHaveBeenCalledWith(expectedResult);
+          expect(mockNextFunction).not.toHaveBeenCalled();
+      });
+    });
+  });
+
+  describe("Given getById method is called", () => {
+    describe("When all the data is correct and no problems are found", () => {
+      it("should search the pagamento and return it", async () => {
+          const mockRequest = {
+              params: {
+                  id: "65a2b665e3efaec673f665bd",
+              },
+          } as any;
+          const mockResponse = {
+              status: jest.fn().mockReturnThis(),
+              json: jest.fn(),
+          } as any;
+          const mockNextFunction = jest.fn();
+
+          const expectedResult = {
+              id: "someId",
+              ...mockRequest.params,
+              pedidoId: "65a2aedd591338377f5a575e",
+              tipo: PagamentoTipoEnum.Aprovado,
+              valorTotal: 36,
+          };
+
+          pagamentoUseCaseMock.getById = jest
+              .fn()
+              .mockResolvedValue(expectedResult);
+
+          await pagamentoController.getById(
+              mockRequest,
+              mockResponse,
+              mockNextFunction,
+          );
+
+          expect(mockResponse.status).toHaveBeenCalledWith(StatusCode.ok);
+          expect(mockResponse.json).toHaveBeenCalledWith(expectedResult);
+          expect(mockNextFunction).not.toHaveBeenCalled();
+      });
+    });
+
+    describe("When the id is not sent through the params", () => {
+      it("it should return unprocessableEntity(422) response error", async () => {

```

Comment:
Aqui só remover o it extra:
```suggestion
      it("should return unprocessableEntity(422) response error", async () => {

```